File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/RoomOptionsViewController.m
//
//  RoomOptionsViewController.m
//  OpenVideoCall
//
//  Created by CavanSu on 2019/6/6.
//  Copyright © 2019 Agora. All rights reserved.
//

#import "RoomOptionsViewController.h"

@implementation RoomOptions
@end

@interface RoomOptionsViewController ()
@property (weak, nonatomic) IBOutlet UISwitch *debugSwitch;
@property (weak, nonatomic) RoomOptions *options;
@end

@implementation RoomOptionsViewController
- (RoomOptions *)options {
    return [self.dataSource roomOptionsVCNeedOptions];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.debugSwitch.on = self.options.isDebugMode;
}

- (IBAction)doDebugPressed:(UISwitch *)sender {
    BOOL debugMode = sender.isOn;
    [self.delegate roomOptions:self debugModeDidEnable:debugMode];
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section {
    return 16;
}

- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section {
    UIView *view = [[UIView alloc] init];
    view.backgroundColor = [UIColor colorWithRed:229.0 / 255.0 green:229.0 / 255.0 blue:229.0 / 255.0 alpha:1];
    return view;
}

@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/RoomOptionsViewController.swift
//
//  RoomOptionsViewController.swift
//  OpenVideoCall
//
//  Created by CavanSu on 2019/5/22.
//  Copyright © 2019 Agora. All rights reserved.
//

import UIKit

struct RoomOptions {
    var isDebugMode: Bool = false
}

protocol RoomOptionsVCDataSource: NSObjectProtocol {
    func roomOptionsVCNeedOptions() -> RoomOptions
}

protocol RoomOptionsVCDelegate: NSObjectProtocol {
    func roomOptionsVC(_ vc: RoomOptionsViewController, debugModeDid enable: Bool)
}

class RoomOptionsViewController: UITableViewController {
    
    @IBOutlet weak var debugSwitch: UISwitch!
    
    weak var delegate: RoomOptionsVCDelegate?
    weak var dataSource: RoomOptionsVCDataSource?
    
    var options: RoomOptions {
        return dataSource!.roomOptionsVCNeedOptions()
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        debugSwitch.isOn = options.isDebugMode
    }
    
    @IBAction func doDebugPressed(_ sender: UISwitch) {
        let debugMode = sender.isOn
        delegate?.roomOptionsVC(self, debugModeDid: debugMode)
    }
}

extension RoomOptionsViewController {
    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 16
    }
}
